R/db-helpers.R

```diff
@@ -47,26 +47,22 @@ create_queries <- function(
 queries_not_nullable <- function(dest, pk_information) {
   db_tables <- pk_information$remote_name
   cols_to_set_not_null <- pk_information$pk_col
-  cols_classes <- pk_information$pk_class
-  cols_db_classes <- class_to_db_class(dest, cols_classes)
 
   if (is_mssql(dest)) {
     pmap_chr(
       list(
         db_tables,
-        cols_to_set_not_null,
-        cols_db_classes
+        cols_to_set_not_null
       ),
-      ~ glue("ALTER TABLE {..1} ALTER COLUMN {..2} {..3} NOT NULL")
+      ~ glue("ALTER TABLE {..1} ALTER COLUMN {..2} NOT NULL")
     )
   } else if (is_postgres(dest)) {
     pmap_chr(
       list(
         db_tables,
-        cols_to_set_not_null,
-        cols_db_classes
+        cols_to_set_not_null
       ),
-      ~ glue("ALTER TABLE {..1} ALTER COLUMN {..2} TYPE {..3}, ALTER COLUMN {..2} SET NOT NULL")
+      ~ glue("ALTER TABLE {..1} ALTER COLUMN {..2} SET NOT NULL")
```

Comment:
for Postgres your approach works.

I am not sure if for MSSQL one can avoid the need for the data type. I checked a few tutorials for rendering columns NOT NULLABLE, but everywhere it seems the data type is used. 

If we do need it for MSSQL we can either:
- do it like it was done before and store the classes in the 'dm' or
- determine the class on the fly when dealing with MSSQL

Reply:
Rewritten to set the `NOT NULL` modifier during creation of the table.